app/models/location.rb

```diff
@@ -2,4 +2,53 @@ class Location < ActiveRecord::Base
   belongs_to :person
   belongs_to :district
   belongs_to :state
+
+  after_save :update_nation_builder
+
+  def update_location(address: nil, city: nil, state: nil, zip: nil)
+    if address
+      if district = District.find_by_address(address: address,
+                                             city:    city,
+                                             state:   state,
+                                             zip:     zip)
+        self.address_1 = address
+        self.city      = city
+        self.district  = district
+        self.state     = district.state
+        self.zip_code  = zip if zip = ZipCode.valid_zip_5(zip)
+      else
+        return nil
+      end
+    elsif zip = ZipCode.valid_zip_5(zip) and zip != self.zip_code
+      zip_code = ZipCode.find_by(zip_code: zip)
+      self.address_1 = nil
+      self.city      = city
+      self.zip_code  = zip
+      self.state     = zip_code.try(:state)
+      self.district  = zip_code.try(:single_district)
+    end
+    self.save
+  end
+
+  def state_abbrev
+    if state
+      state.abbrev
+    elsif district
+      district.state.abbrev
+    end
+  end
+
+  private
+
+  def serializable_hash(options)
```

Comment:
Unused method argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used. You can also write as `serializable_hash(*)` if you want the method to accept any arguments but don't care about them.